src/volume.ts

```diff
@@ -359,7 +359,15 @@ export class Volume implements FsCallbackApi, FsSynchronousApi {
       throw new Error('createLink: name cannot be empty');
     }
 
-    return parent.createChild(name, this.createNode(isDirectory, perm));
+    // If no explicit permission is provided, use defaults based on type
+    const defaultPerm = isDirectory ? 0o777 : 0o666;
+    const finalPerm = mode ?? defaultPerm;
```

Comment:
nit: I don't mind this being inlined since it's the only place we use the variable, and we've got a comment that explicitly calls out that we're got a default so I don't think it'll be easy to misunderstand

```suggestion
    const finalPerm = mode ?? (isDirectory ? 0o777 : 0o666);
```